usr.sbin/devinfo/devinfo.c

```diff
@@ -200,65 +364,115 @@ print_rman_resource(struct devinfo_res *res, void *arg __unused)
 int
 print_rman(struct devinfo_rman *rman, void *arg __unused)
 {
-	printf("%s:\n", rman->dm_desc);
+	char*	safe_desc;
+	size_t	len;
+
+	len = strlen(rman->dm_desc) + 1;
+	if ((safe_desc = malloc(len)) == NULL) {
+		xo_err(1, "No memory!");
+	}
+	strlcpy(safe_desc, rman->dm_desc, len);
+	xml_safe_string(safe_desc);
```

Comment:
This is the third instance of this, that suggests a common function and a little refactoring would be in order.

Reply:
Fixed by putting more logic into xml_safe_string(). Shaved of 15 lines in total. Thanks.